src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -508,11 +509,14 @@ def project_statement(
     child: str,
     is_distinct: bool = False,
     child_uuid: Optional[str] = None,
+    ilike_pattern: Optional[str] = None,
 ) -> str:
     if not project:
         columns = STAR
     else:
         columns = NEW_LINE + TAB + (COMMA + NEW_LINE + TAB).join(project)
+    if ilike_pattern:
+        columns = f"{columns}{ILIKE}{SINGLE_QUOTE}{ilike_pattern}{SINGLE_QUOTE}"
```

Comment:
does it work for non select * cases as well?

Reply:
no it doesn't work for non-select *, otherwise it would be selecting by filtering value `select a ilike '%somevalue%'`.
I have moved the logic under the "not project" condition.

based upon current implementation, it should not hit the else condition, I added an assertion there